src/react/hooks/useReadQuery.ts

```diff
@@ -36,32 +38,23 @@ export interface UseReadQueryResult<TData = unknown> {
 export function useReadQuery<TData>(
   queryRef: QueryReference<TData>
 ): UseReadQueryResult<TData> {
-  const internalQueryRef = unwrapQueryRef(queryRef);
-  invariant(
-    internalQueryRef.promiseCache,
-    "It appears that `useReadQuery` was used outside of `useBackgroundQuery`. " +
-      "`useReadQuery` is only supported for use with `useBackgroundQuery`. " +
-      "Please ensure you are passing the `queryRef` returned from `useBackgroundQuery`."
+  const [internalQueryRef, getPromise] = React.useMemo(
+    () => unwrapQueryRef(queryRef),
+    [queryRef]
   );
 
-  const { promiseCache, key } = internalQueryRef;
-
-  if (!promiseCache.has(key)) {
-    promiseCache.set(key, internalQueryRef.promise);
-  }
-
-  const promise = useSyncExternalStore(
+  let promise = useSyncExternalStore(
```

Comment:
```suggestion
  const promise = useSyncExternalStore(
```

Perhaps I'm missing where its reassigned, but if not, could we use a `const` here?

Reply:
It was before, but isn't anymore :) Good catch!